daemon/emer-persistent-cache.c

```diff
@@ -501,121 +431,99 @@ compute_boot_offset (EmerPersistentCache *self,
     {
       if (always_update_timestamps)
-        {
-          gboolean write_success =
-            save_timing_metadata (self, &relative_time, &absolute_time,
-                                  NULL, NULL, NULL, &error);
-          if (!write_success)
-            {
-              g_warning ("Failed to update relative and absolute time in "
-                         "metadata file. Error: %s.", error->message);
-              g_error_free (error);
-            }
-        }
+        return save_timing_metadata (self, &relative_time, &absolute_time, NULL,
+                                     NULL, NULL, error);
 
       return TRUE;
     }
 
-  if (!g_key_file_load_from_file (priv->boot_offset_key_file,
-                                  priv->boot_metadata_file_path,
-                                  G_KEY_FILE_NONE,
-                                  &error))
+  GError *local_error = NULL;
+  gboolean load_succeeded =
+    g_key_file_load_from_file (priv->boot_offset_key_file,
+                               priv->boot_metadata_file_path, G_KEY_FILE_NONE,
+                               &local_error);
+  if (!load_succeeded)
     {
-      if (!g_error_matches (error, G_KEY_FILE_ERROR,
+      if (!g_error_matches (local_error, G_KEY_FILE_ERROR,
                             G_KEY_FILE_ERROR_NOT_FOUND) &&
-          !g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
-        g_warning ("Got an unexpected error trying to load %s. Error: %s.",
-                   priv->boot_metadata_file_path, error->message);
+          !g_error_matches (local_error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+        {
+          g_propagate_error (error, local_error);
+          return FALSE;
+        }
 
-      g_error_free (error);
       return reset_boot_offset_metadata_file (self, &relative_time,
-                                              &absolute_time);
+                                              &absolute_time, error);
     }
 
-  gint64 boot_offset = g_key_file_get_int64 (priv->boot_offset_key_file,
-                                             CACHE_TIMING_GROUP_NAME,
-                                             CACHE_BOOT_OFFSET_KEY, &error);
-  if (error != NULL)
+  gint64 boot_offset =
+    g_key_file_get_int64 (priv->boot_offset_key_file, CACHE_TIMING_GROUP_NAME,
+                          CACHE_BOOT_OFFSET_KEY, &local_error);
+  if (local_error != NULL)
     {
-      g_warning ("Could not find a valid boot offset in the metadata file. "
-                 "Error: %s.", error->message);
-      g_error_free (error);
-      return reset_boot_offset_metadata_file (self, &relative_time,
-                                              &absolute_time);
+      g_propagate_prefixed_error (error, local_error,
+                                  "Failed to read boot offset from metadata "
+                                  "file %s. ",
+                                  priv->boot_metadata_file_path);
+      return FALSE;
```

Comment:
Yeah, this change belongs in a separate commit along with the one below. I didn't catch on to this when I reviewed the previous code originally.